File: Hot4/Hot4.Service/Concrete/ChannelFactoryWrapper.cs
using Hot4.Service.Abstract;
using System;
using System.Collections.Generic;
using System.Linq;
using System.ServiceModel;
using System.Text;
using System.Threading.Tasks;
using ZesaAPI;

namespace Hot4.Service.Concrete
{
    public class ChannelFactoryWrapper<T> : INetworkClient<T>
    {
        private readonly ChannelFactory<T> _factory;

        public ChannelFactoryWrapper(ChannelFactory<T> factory)
        {
            _factory = factory;
        }
        public void Abort()
        {
            _factory.Abort();
        }

        public void Close()
        {
            _factory.Close();
        }

        public T GetNetwork()
        {
            return _factory.CreateChannel();
        }

    }
}

File: Customer APIs/hot.backend.all/hot.core/hot.core/ServiceWrappers/ChannelFactoryWrapper.vb
Imports System.ServiceModel
Imports Hot.Core.EconetPay

Namespace ServiceWrappers
    Public Class ChannelFactoryWrapper (Of T)
        Implements INetworkClient(Of T)

        Private ReadOnly _factory As ChannelFactory(Of T)

        Public Sub New(factory As ChannelFactory(Of T))
            _factory = factory
        End Sub

        Public Function GetNetwork() As T Implements INetworkClient(Of T).GetNetwork
            Return _factory.CreateChannel()
        End Function

        Public Sub Close() Implements INetworkClient(Of T).Close
            _factory.Close()
        End Sub

        Public Sub Abort() Implements INetworkClient(Of T).Abort
            _factory.Abort()
        End Sub     
    End Class

End NameSpace